web-client/public/js/graph.js

```diff
@@ -1101,11 +1101,12 @@ export var drawGraph = function (workbook) {
                                 average
                             );
                             timePoints = result.timePoints;
-                            return result.data;
-                        } else {
-                            return 0;
+                            return result.data && result.data.length > 0
+                                ? result.data
+                                : [];
```

Comment:
Looking to tighten up the code here, it seems that the case of `result.data.length === 0` means that `result.data` is already `[]` so making it branch to return a hardcoded `[]` seems redundant

From the look of it, you really just need to return the fixed `[]` if `result.data` is falsy, right? Would this work?
```suggestion
                            return result.data ?? [];
```

Reply:
I tried this approach before, and I got a parser error.

Reply:
Ah OK—that probably means our GRNsight parser hasn’t been updated to support `??`

Earlier versions of JavaScript used `||`—it should generally still work here:
```javascript
                            return result.data || [];
```

Reply:
Yes, it works! `&&` is used for when `result.data` is null or undefined. But for `||`, it will return `[]` if `result.data` is null, undefined, 0, or "".